src/kudu/consensus/raft_consensus.cc

```diff
@@ -2534,13 +2534,26 @@ Status RaftConsensus::RequestVoteRespondVoteGranted(const VoteRequestPB* request
   MonoDelta backoff = LeaderElectionExpBackoffDeltaUnlocked();
   SnoozeFailureDetector(string("vote granted"), backoff);
 
+  // Get previous vote history here since it gets updated
+  // by `SetVotedForCurrentTermUnlocked`.
+  const std::map<int64_t, PreviousVotePB>& previous_vote_history =
```

Comment:
seems like RequestVoteRespondVoteGranted is called under raft consensus lock.
One suggestion is to put a DCHECK for lock in that function.

Because of that this strategy of reference is probably safe, but risky for other users of
the same function. So we need warnings atleast.

Reply:
I will add a DCHECK for the lock.